### PR TITLE
rmw 0.7.09 (new formula)

### PR DESCRIPTION
**Pull request by andy5995**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/rmw.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  head "https://github.com/theimpossibleastronaut/rmw.git"
+
+  depends_on "ncurses"
+
+  def install
+    system  "./configure", "--enable-debug=no",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system  "make", "install"
```
Comment (nandahkrishna) [inline Formula/rmw.rb:16]:
```suggestion
    system "./configure", "--enable-debug=no",
                          "--disable-dependency-tracking",
                          "--disable-silent-rules",
                          "--prefix=#{prefix}"
    system "make", "install"
```

File: Formula/rmw.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  test do
+    (testpath/"foo").write("bar")
+    system "#{bin}/rmw", (testpath/"foo")
+    system "#{bin}/rmw", "-u"
+    system "#{bin}/rmw", "-l"
+    system "#{bin}/rmw", "-vvg"
```
Comment (nandahkrishna) [inline Formula/rmw.rb:24]: Is it possible to compare some output here? Also, you could create a file, remove it and check that it doesn't exist.

**Review by nandahkrishna — COMMENTED** [review]
Thanks for your PR @andy5995, I have some suggestions. Once you've made these changes, please squash all your commits into one commit, with the message:
```
rmw 0.7.09 (new formula)
```

File: Formula/rmw.rb (hunk 2 above)
Comment (andy5995) [reply Formula/rmw.rb]: Thanks for reviewing @nandahkrishna . I added a check for the file existence like you suggested.

I could compare some output using substrings, except for one minor problem. I don't know ruby.

This is the output, which is all correct. Any suggestions on how I can compare output of those "system" commands by using substrings?

```
qandy@oceanus:~/src/homebrew-core/Formula$ brew test --verbose rmw.rb 
==> Testing rmw
==> /home/linuxbrew/.linuxbrew/Cellar/rmw/0.7.09/bin/rmw /tmp/rmw-test-20210513-22759-pusr7f/foo
Created directory /tmp/rmw-test-20210513-22759-pusr7f/.local/share/rmw
Created directory /tmp/rmw-test-20210513-22759-pusr7f/.config
Creating default configuration file:
  /tmp/rmw-test-20210513-22759-pusr7f/.config/rmwrc

Created directory /tmp/rmw-test-20210513-22759-pusr7f/.local/share/Waste/files/
Created directory /tmp/rmw-test-20210513-22759-pusr7f/.local/share/Waste/info/
1 file was removed to the waste folder
PASS (file not found)
==> /home/linuxbrew/.linuxbrew/Cellar/rmw/0.7.09/bin/rmw -u
+'/tmp/rmw-test-20210513-22759-pusr7f/.local/share/Waste/files/foo' -> '/tmp/rmw-test-20210513-22759-pusr7f/foo'
PASS (file exists again)
==> /home/linuxbrew/.linuxbrew/Cellar/rmw/0.7.09/bin/rmw -l
/tmp/rmw-test-20210513-22759-pusr7f/.local/share/Waste
==> /home/linuxbrew/.linuxbrew/Cellar/rmw/0.7.09/bin/rmw -vvg
PATH_MAX = 4096
RMW_FAKE_MEDIA_ROOT:false
config file: /tmp/rmw-test-20210513-22759-pusr7f/.config/rmwrc
purging is disabled ('purge_after' is set to '0')

most recent list (mrl file): /tmp/rmw-test-20210513-22759-pusr7f/.local/share/rmw/mrl
```

Comment (andy5995) [reply Formula/rmw.rb]: > This is the output, which is all correct. Any suggestions on how I can compare output of those "system" commands by using substrings?
> 

@nandahkrishna ok, I figured out some ruby and pushed all my new changes. Thank you for the suggestions.

File: Formula/rmw.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  test do
+    (testpath/"foo").write("bar")
+    assert_match "removed", shell_output("#{bin}/rmw #{testpath}/foo")
+    assert !File.file?(testpath/"foo")
+    system "#{bin}/rmw", "-u"
+    assert File.file?(testpath/"foo")
+    assert_match "/.local/share/Waste", shell_output("#{bin}/rmw -l")
+    assert_match "purging is disabled", shell_output("#{bin}/rmw -vvg")
```
Comment (nandahkrishna) [inline Formula/rmw.rb:26]:
```suggestion
    file = testpath/"foo"
    touch file
    assert_match "removed", shell_output("#{bin}/rmw #{file}")
    refute_predicate file, :exist?
    system "#{bin}/rmw", "-u"
    assert_predicate file, :exist?
    assert_match "/.local/share/Waste", shell_output("#{bin}/rmw -l")
    assert_match "purging is disabled", shell_output("#{bin}/rmw -vvg")
```

**Review by nandahkrishna — COMMENTED** [review]
Nearly there, just a suggestion to improve the test.

**Comment by andy5995** [comment]
I changed the depends line. Could someone trigger another workflow run?

**Comment by andy5995** [comment]
Hmm.. the PR run failed again. The ncurses menu library can't be found. I might be able to fix it by preceding the configure script with PKG_CONFIG_PATH= something but not sure where the menu library would be located. Any ideas?

```
 checking for ncursesw via pkg-config... yes
checking for working ncursesw/curses.h... no
checking for working ncursesw.h... no
checking for working ncurses.h... yes
checking for Curses Menu library with ncursesw/menu.h... no
checking for free_menu in -lmenu... no
configure: error: requires the Curses menu library.
```

**Comment by nandahkrishna** [comment]
It seems like the `ncurses` shipped with macOS doesn't come with the `menu` library, so you'll have to use Homebrew `ncurses` instead (`depends_on`, just as it was earlier). As for the audit failure, I think there's an allowlist in `audit_exceptions/` you should add the formula to.

**Comment by carlocab** [comment]
You can do `depends_on "ncurses"`. The audit complaining about `depends_on` instead of `uses_from_macos` is only triggered for new formulae.

Can we check that this doesn't exist with macOS `ncurses`? I won't be surprised if it doesn't, because macOS `ncurses` is over 15 years old.

**Comment by andy5995** [comment]
> You can do `depends_on "ncurses"`. The audit complaining about `depends_on` instead of `uses_from_macos` is only triggered for new formulae.
> 
> Can we check that this doesn't exist with macOS `ncurses`? I won't be surprised if it doesn't, because macOS `ncurses` is over 15 years old.

Ok, I made the change. Can we trigger another build?

File: Formula/rmw.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  homepage "https://remove-to-waste.info/"
+  url "https://github.com/theimpossibleastronaut/rmw/releases/download/v0.7.09/rmw-0.7.09.tar.gz"
+  sha256 "d0a85944d03e7ec2a94b25d6d6ac92880fd0d3f63d90bb9ed56b16418fd41c69"
+  license "GPL-3.0-or-later"
+  head "https://github.com/theimpossibleastronaut/rmw.git"
+
+  depends_on "ncurses"
+
+  def install
+    ENV["PKG_CONFIG_PATH"] = "/usr/local/opt/ncurses/lib/pkgconfig"
```
Comment (carlocab) [inline Formula/rmw.rb:12]: Can we build without this? You don't actually have a `pkg-config` dependency, so I'm pretty sure this is not used. We also shouldn't be hardcoding a `/usr/local` prefix here.

File: Formula/rmw.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  homepage "https://remove-to-waste.info/"
+  url "https://github.com/theimpossibleastronaut/rmw/releases/download/v0.7.09/rmw-0.7.09.tar.gz"
+  sha256 "d0a85944d03e7ec2a94b25d6d6ac92880fd0d3f63d90bb9ed56b16418fd41c69"
+  license "GPL-3.0-or-later"
+  head "https://github.com/theimpossibleastronaut/rmw.git"
+
+  depends_on "ncurses"
+
+  def install
+    ENV.append_path "PKG_CONFIG_PATH", Formula["ncurses"].opt_lib/"pkgconfig"
```
Comment (carlocab) [inline Formula/rmw.rb:12]: What happens when you build without this? `ncurses`' `pkgconfig` directory should already be in `PKG_CONFIG_PATH` without doing this.

File: Formula/rmw.rb (hunk 4 above)
Comment (andy5995) [reply Formula/rmw.rb]: > Can we build without this? You don't actually have a `pkg-config` dependency, so I'm pretty sure this is not used. We also shouldn't be hardcoding a `/usr/local` prefix here.

I changed it to `ENV.append_path "PKG_CONFIG_PATH", Formula["ncurses"].opt_lib/"pkgconfig"`
> 
> `brew` also automatically specifies `PKG_CONFIG_PATH` correctly based on dependencies.

I've always needed it when building this program on Travis, using brew to install ncurses. And I just set it up on GitHub actions. First time I tried without using PKG_CONFIG_PATH, it failed. I added it, and it worked.

File: Formula/rmw.rb (hunk 5 above)
Comment (andy5995) [reply Formula/rmw.rb:12]: I can't try directly as I don't have a macOS computer. But as I noted in my last comment, the build has always failed for me on Travis and more recently, using GitHub actions.

Comment (andy5995) [reply Formula/rmw.rb:12]: > What happens when you build without this? `ncurses`' `pkgconfig` directory should already be in `PKG_CONFIG_PATH` without doing this.

Here's an example of it working when PKG_CONFIG_PATH is used before configure, but then not for 'make distcheck'

https://github.com/theimpossibleastronaut/rmw/runs/2579079173?check_suite_focus=true

And here, when it's added, it completes successfully.
https://github.com/theimpossibleastronaut/rmw/runs/2579116775?check_suite_focus=true

Comment (carlocab) [reply Formula/rmw.rb:12]: You are running that outside of `brew`, though, so you don't benefit from `brew` setting env variables correctly for the build. This build is running inside of `brew`, so `brew` should be setting these env variables.

File: Formula/rmw.rb (hunk 4 above)
Comment (carlocab) [reply Formula/rmw.rb]: As mentioned in my other comment, your Travis build runs outside `brew`, so of course it doesn't benefit from `brew` setting the env variables correctly. We definitely don't need this. Please remove it.

File: Formula/rmw.rb (hunk 5 above)
Comment (andy5995) [reply Formula/rmw.rb]: Ok, I've removed it.

**Review by carlocab — APPROVED** [review]
I've made a small revision to document the choice to use brewed `ncurses` instead of the system one.

Thanks, @andy5995!

**Comment by andy5995** [comment]
Thank you for the help

**Pull request closed without merging**